cmd/wio/commands/run/util.go

```diff
@@ -2,29 +2,52 @@ package run
 
 import (
     "os"
+    "path/filepath"
     "wio/cmd/wio/commands/run/cmake"
     "wio/cmd/wio/constants"
+    "wio/cmd/wio/errors"
     "wio/cmd/wio/types"
     "wio/cmd/wio/utils/io"
 )
 
+var sourceExtensions = map[string]bool{
+    ".cpp": true,
+    ".c":   true,
+    ".cc":  true,
+}
+
+// perform a check that at least one executable file is in target directory
+func sourceFilesExist(directory string) (bool, error) {
+    success := "success"
+    err := filepath.Walk(directory, func(path string, f os.FileInfo, err error) error {
```

Comment:
again, we're not trying to replace Cmake, if there are errors with source files then let cmake generate the errors.